### DISCUSSION TITLE
zettlr doesn't open any file

### DISCUSSION THREAD

**Discussion by DoctorSubtilis** [Zettlr/Zettlr#4814] [General]
In my KDE Neon (latest release) when I click on a md file, Zettlr starts, but without open _that_ file.  
It seems back to windows 3.1 (before windows 95), when you are able to open not _objects_, but only _programs_ (/apps), and only _within them_ open single files.  
It's my fault? It is possible to fix this?

**Comment by GrippenDynamik** [comment]
I had the same problem with:
- Linux Mint 21.3 with Xfce 64-bit
- Zettlr 3.0.5 stable (from Flathub)

The problem was in the file:
/var/lib/flatpak/exports/share/applications/com.zettlr.Zettlr.desktop
In the line:
Exec=/usr/bin/flatpak run --branch=stable --arch=x86_64 --command=start-zettlr com.zettlr.Zettlr

The solution was to change it to:
Exec=/usr/bin/flatpak run --branch=stable --arch=x86_64 --command=start-zettlr com.zettlr.Zettlr @@ %u @@

**Reply by nathanlesage** [reply]
@lunarequest Might this be some improvement idea for the flatpak release?

**Reply by obitwrite** [reply]
I'm actually experiencing something similar with opening files, but on macOS. When Zettlr is already open and I click a file in Finder, it doesn't bring Zettlr to the forefront. I have to click on Zettlr in the Dock first. See this issue: https://github.com/Zettlr/Zettlr/issues/4755. Great that the flatpak issue was solved!

**Comment by GrippenDynamik** [comment]
When I opened a markdown file in Thunar today, Zettlr was back to its old behavior.
Zettlr opened, but didn't show the file.
So my solution above was incomplete.

But this one is probably correct:
Exec=/usr/bin/flatpak run --branch=stable --arch=x86_64 --command=start-zettlr --file-forwarding com.zettlr.Zettlr @@ %F @@

I copied it from the [MarkText](https://www.marktext.cc/) launcher.
- "--file-forwarding" ensures that Zettlr can read and write to the file(s), see the [Flatpak Command Reference](https://docs.flatpak.org/en/latest/flatpak-command-reference.html)
- "%F" means multiple file paths, see [The Exec key](https://specifications.freedesktop.org/desktop-entry-spec/latest/ar01s07.html)
"%U" would probably also work, because it allows file paths and URLs. But can Zettlr open remote URLs?
